### ISSUE TITLE
EventRelay End-to-End: Remaining Work After Pipeline Fix

### ISSUE THREAD

**Issue by groupthinking** [groupthinking/EventRelay#139]
## Context

Commit `735df96b` wired the EventRelay pipeline end-to-end: CloudEvents emission, SkillBuilder learning loop, training data collection for Vertex AI fine-tuning, and several bug fixes. Tested with 5 diverse technical videos — 4/5 pass.

This issue tracks the remaining work to make the pipeline flawless for all video types and connect the async infrastructure that already exists in the repo.

---

### 1. Wire async long-video processing (High Priority)
**Problem:** Videos >30 min timeout on the synchronous `transcript-action` endpoint (180s HTTP limit).

**Existing code (not wired):**
- `src/youtube_extension/services/cloud/cloud_tasks_queue.py` — Cloud Tasks queue with retry + backoff
- `src/youtube_extension/services/cloud/cloud_video_processor.py` — Cloud-native processor (Firestore state + Cloud Tasks + Vertex AI)
- `src/integration/temporal_video_analysis.py` — Temporal segment-by-segment analysis
- `apps/web/src/app/api/pipeline/stream/route.ts` — SSE streaming for real-time progress

**Fix:** Route long videos (>15 min) through `CloudTasksQueueService` → process async → stream results via SSE. Return a job ID immediately, let the client poll or stream.

### 2. Playlist URL validation (Low Priority)
**Problem:** Playlist URLs don't contain a video ID. The metadata fetcher tries all APIs and times out instead of failing fast.

**Fix:** Add URL validation in `TranscriptActionWorkflow.run()` — reject playlist URLs with a clear error before hitting the YouTube API.

### 3. Wire Ray Serve models into pipeline (Medium Priority)
**Problem:** `TranscriptQualityScorer` and `ActionPriorityRanker` exist in `src/uvai/ml/serve.py` with outcome recording endpoints, but the pipeline never calls them.

**What to wire:**
- Before transcript extraction: call `/score-transcript` to pick optimal source
- After extraction: call `/score-transcript/outcome` with actual results
- After action generation: call `/rank-actions` to prioritize outputs
- On user interaction: call `/rank-actions/feedback` with click/completion data

### 4. Phase 2 model weight updates (Medium Priority)
**Problem:** Both Ray Serve models say "Phase 1: Just log. Phase 2: Update weights via gradient" but Phase 2 isn't implemented.

**Files:**
- `src/uvai/ml/models/transcript_quality_scorer.py:401` — `record_outcome()` just logs
- `src/uvai/ml/models/action_priority_ranker.py:317` — `record_feedback()` just logs

### 5. BigQuery export for training data (Low Priority)
**Problem:** SkillBuilder stores lessons locally in `~/.uvai/skills/`. For the Vertex AI fine-tuning loop, accumulated outcomes should also flow to BigQuery.

**Existing code:**
- `scripts/archive/tree_to_bigquery.py` — BigQuery client for project `uvai-730bb`
- `apps/web/src/lib/training-store.ts` — JSONL accumulator (100 example threshold)
- `apps/web/src/app/api/training/trigger/route.ts` — Uploads to GCS + submits Vertex AI SFT job

### 6. Dependabot vulnerabilities
41 vulnerabilities flagged on the default branch (1 critical, 21 high). Review at: https://github.com/groupthinking/EventRelay/security/dependabot

---

## What's Already Working (from commit 735df96b)

| Component | Status |
|-----------|--------|
| Pipeline (transcript-action) | 3-agent analysis (summary, personality, strategy) |
| CloudEvents emission | received/completed/failed events at each stage |
| SkillBuilder learning | Context injection + outcome recording |
| Training data collection | saveTrainingExample → JSONL → Vertex AI trigger |
| APIKeyMiddleware | Fixed import name mismatch |
| Gemini fallback | Fixed genai_types.Part/Content/FileData patch |
| Metadata fallback | Graceful degradation when YouTube Data API fails |
| Dockerfile | Non-editable install, no ML bloat |
| Input validation | SQL injection, XSS, malformed URLs all handled |
| Concurrent requests | 2 simultaneous requests work without interference |

## Test Evidence

```
Phase 1 (Health):      4/4 PASS
Phase 2 (Validation):  6/6 PASS
Phase 3 (Pipeline):    1/1 PASS
Phase 4 (Edge Cases):  3/4 PASS (playlist timeout)
Phase 5 (Concurrent):  2/2 PASS
Phase 6 (Endpoints):   4/4 PASS
Phase 7 (Side Effects): 22 CloudEvents, 3 SkillBuilder skills
```
